main.tf

```diff
@@ -7,8 +7,8 @@
 locals {
   # ibm_container_vpc_cluster automatically names default pool "default" (See https://github.com/IBM-Cloud/terraform-provider-ibm/issues/2849)
   default_pool            = element([for pool in var.worker_pools : pool if pool.pool_name == "default"], 0)
-  other_pools             = [for pool in var.worker_pools : pool if pool.pool_name != "default" && !var.ignore_worker_pool_size_changes]
-  other_autoscaling_pools = [for pool in var.worker_pools : pool if pool.pool_name != "default" && var.ignore_worker_pool_size_changes]
+  other_pools             = coalesce(local.default_pool.import_on_create, false) ? [for pool in var.worker_pools : pool if !var.ignore_worker_pool_size_changes] : [for pool in var.worker_pools : pool if pool.pool_name != "default" && !var.ignore_worker_pool_size_changes]
```

Comment:
I suggest to change the name of this variable as this list can now contains the default pool based on the updated logic in the case where import on create is set to true, so the name becomes is confusing.

[for pool in var.worker_pools : pool if !var.ignore_worker_pool_size_changes]